### DISCUSSION TITLE
[FEATURE REQ] Auto add FL Token to download link (Append URL parameters)

### DISCUSSION THREAD

**Discussion by kasmador** [autobrr/autobrr#1658] [Ideas]
**Is your feature request related to a problem? Please describe.**
When downloading a torrent using a free leech token, some torrent sites just require using the same download URL with the `&freeleech=1` URL parameter added to the end.

I would like the option to enable automatically turning a download into a free leech by using said tokens.

**Describe the solution you'd like**
Add a section to the Advanced tab of of the Filter configuration which allows adding key/value pairs which will be added as URL parameters to the download link before it gets downloaded.

**Describe alternatives you've considered**
To make it even more seamless, when adding an indexer add a definition of how to add using a free leech token as part of the indexer config. This would then turn it into a simple toggle inside which can be added into the Advanced tab.

**Additional context**
Useful for those low volume, consistent, large files that you want to download.

**Comment by adityaxdiwakar** [comment]
I would love to see something like this. :+1:

**Comment by nebb00** [comment]
Would love this! Assume no current workaround to make this work

**Comment by mfin** [comment]
I would love this! A couple of trackers also use `&usetoken=1`.

**Comment by m7ke** [comment]
Was just researching to see if this was possible, I would also really like to see this!

**Comment by STT414** [comment]
Please add this, it's something you can do with autodl-irssi.

**Comment by zze0s** [comment]
> Please add this, it's something you can do with autodl-irssi.

Is that so? Do you have any docs or sourcecode for that feature?

**Comment by STT414** [comment]
> > Please add this, it's something you can do with autodl-irssi.
> 
> Is that so? Do you have any docs or sourcecode for that feature?

I believe it was more of a workaround than a feature. I seem to remember it involved simply appending `&freeleech=1` into the tracker config. Unfortunately I lost my config files and this was something I setup many years ago.

Now that I think about it, the same thing might work with autobrr by appending it to the torrent pass field.

**Comment by zze0s** [comment]
Thought so. That is indeed possible to do in autobrr as well by appending the correct urlparam to the passkey/key field in indexer settings. But just note that it's not ideal and you might use up tokens when you don't expect it.

For indexers that do not announce size for example, and when you have a min/max size in the filter it will download the .torrent to parse size and that will then use your token even if you might not actually download the file. We have API check for PTP, GGn, RED, OPS and BTN so it does not apply there.
